File: PHP/Hardware.php
<?php

/* inherit class */
class Hardware extends Product{

	/* atribut private */
	private $brand = "";
	private $model = "";

	/* konstruktor */
	public function __construct($brand = "", $model = ""){
		$this->brand = $brand;
		$this->model = $model;
	}

	/* setter dan getter untuk atribut brand */
	public function setBrand($brand){
		$this->brand = $brand;
	}

	public function getBrand(){
		return $this->brand;
	}

	/* setter dan getter untuk atribut model */
	public function setModel($model){
		$this->model = $model;
	}

	public function getModel(){
		return $this->model;
	}

	/* method untuk mencetak output atribut dari class Product */
	public function printHardware(){
		echo "Merk           : ".$this->getBrand()."<br>";
		echo "Model          : ".$this->getModel()."<br>";
	}
}

?>

File: C++/Hardware.cpp
#include <iostream>
#include <string>

/* import file */
#include "Product.cpp"

using namespace std;

/* inherit class */
class Hardware : public Product{

	/* atribut private */
	private:
			string brand;
			string model;

	/* atribut public */
	public:

		/* konstruktor */
		Hardware(){}

		/* setter dan getter untuk atribut brand */
		void setBrand(string brand){
			this->brand = brand;
		}

		string getBrand(){
			return this->brand;
		}

		/* setter dan getter untuk atribut model */
		void setModel(string model){
			this->model = model;
		}

		string getModel(){
			return this->model;
		}

		/* method untuk mencetak output atribut dari class Hardware */
		void printHardware(){
			cout << "Merk           : " << this->brand << endl;
			cout << "Model          : " << this->model << endl;;
		}

		/* destruktor */
		~Hardware(){}
};